lewis/adapters/modbus.py

```diff
@@ -307,7 +308,9 @@ def process(self, data):
                 'Request: %s', str(['{:#04x}'.format(c) for c in request.to_bytearray()]))
 
             handler = self._get_handler(request.fcode)
-            response = handler(request)
+
+            with device_lock:
+                response = handler(request)
```

Comment:
Same idea as above, I think the lock should go around the request loop.

Don't see any easy way to optimize by splitting device IO from network IO here. Should be fine though.